cmake/CompilerFlags.cmake

```diff
@@ -90,12 +95,29 @@ elseif(CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" O
     target_compile_options(project_warnings INTERFACE -Wno-invalid-source-encoding)
   endif()
 
+  # TODO: after we fix all test, enable this by default on Debug builds
+  # set(need_arithm_debug_genex "$<OR:$<BOOL:${FORCE_DEBUG_ARITHM_GCC_OR_CLANG}>,$<CONFIG:Debug>>")
+  set(need_arithm_debug_genex "$<BOOL:${FORCE_DEBUG_ARITHM_GCC_OR_CLANG}>")
```

Comment:
Disable Floating point exceptions in Debug mode on Unix (clang/gcc) for now until all NaN in unit tests are fixed